src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsUapHiViMobileWiki.java

```diff
@@ -0,0 +1,274 @@
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.AssertionAds;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.HiViUap;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.helpers.SoundMonitor;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+@InBrowser(
+  browser = Browser.CHROME_MOBILE,
+  emulator = Emulator.GOOGLE_NEXUS_5
+)
+@Test(groups = "AdsUapHiViMobileWiki")
+public class TestAdsUapHiViMobileWiki extends TemplateNoFirstLoad {
+  private static final double IMPACT_STATE_ASPECT_RATIO = 272.0 / 153.0;
+  private static final double RESOLVED_STATE_ASPECT_RATIO = 640.0 / 213;
+  private static final String TLB_SLOT_ID = "MOBILE_TOP_LEADERBOARD";
+  private static final By TLB_SELECTOR = By.id(TLB_SLOT_ID);
+  private static final String AD_REDIRECT = "http://fandom.wikia.com/articles/legacy-luke-skywalker";
+  private static final Cookie NO_SMART_BANNER_COOKIE = new Cookie("fandom-sb-closed", "1");
+
+  private AdsBaseObject openPage() {
+    final Page mainPage = new Page("project43", "Project43_Wikia");
+    final AdsBaseObject adsBaseObject = new AdsBaseObject(driver, mainPage.getUrl());
+    driver.manage().addCookie(NO_SMART_BANNER_COOKIE); // it must be run until ADEN-6608 will be released
+    adsBaseObject.getUrl(AdsDataProvider.UAP_HIVI_PAGE.getUrl());
+    return adsBaseObject.waitForPageLoaded();
+  }
+
+  private AdsBaseObject openPageOnSecondPV() {
+    final AdsBaseObject adsBaseObject = openPage();
+    adsBaseObject.refreshPage();
+    final WebElement slot = adsBaseObject.wait.forElementPresent(TLB_SELECTOR);
+    adsBaseObject.waitForSlotExpanded(slot);
+    return adsBaseObject;
+  }
+
+  @Test
+  public void shouldHaveCorrectAspectRatioForImpactState() {
+    AdsBaseObject page = openPage();
+    final WebElement slot = page.wait.forElementPresent(TLB_SELECTOR);
+    page.waitForSlotExpanded(slot);
+    AssertionAds.assertAspectRatio(slot.getSize(), IMPACT_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldHaveCorrectAspectRatioForResolvedState() {
+    openPageOnSecondPV();
+    final WebElement slot = (new HiViUap(driver, TLB_SLOT_ID)).waitForAdLoaded();
+    AssertionAds.assertAspectRatio(slot.getSize(), RESOLVED_STATE_ASPECT_RATIO);
+  }
+
+  @Test
+  public void shouldHaveResolvedStateAfterScroll() throws InterruptedException {
+    AdsBaseObject adsPage = openPage();
+    WebElement slot = driver.findElement(TLB_SELECTOR);
+    int defaultStateHeight = slot.getSize().getHeight();
+    int scrollBy = 50;
+
+    AssertionAds.assertAspectRatio(slot.getSize(), IMPACT_STATE_ASPECT_RATIO);
+
+    adsPage.scrollBy(0, scrollBy);
+    TimeUnit.SECONDS.sleep(1);
```

Comment:
Why do the test sleep here? ;) Is it because it waits for viewability to be counted?

Reply:
No, it waits for correct aspect ratio. Without it, selenium sometimes can't catch correct size of element (it reads size too fast - before resize).